source/getting-started/tutorials/timing-manager/index.md

```diff
@@ -0,0 +1,309 @@
+- User set `TASK_UPDATES_PER_SEC`
+- PWM Frequency
+- User Event Ratio
+- Sensor collection time (`Post-Sensor Mode`)
+- Control task time (how long it takes for the control task to run)
+
+This figure illustrates the functionality of the timing manager:
+
+ ![Timing Manager Architecture](/firmware/arch/images/timing.png)
+
+Read [this docs page](/firmware/arch/timing-manager.md) for detailed information on the Timing Manager.
+
+Consider: Control tasks can run, at most, once every `X` PWM periods, where `X` is the User Event Ratio.\
+That means we have to satisfy these three inequalities:
+
+$$
+\frac{\rm PWM\ frequency}{\rm event\ ratio} \ge \rm TASK\_UPDATES\_PER\_SEC \\
+\\
+\frac{\rm event\ ratio}{\rm PWM\ frequency} \ge \rm Sensor\ Collection\ Time \\
+\\
+\frac{1}{\rm TASK\_UPDATES\_PER\_SEC} < \rm Control\ Task\ Time
+$$ (eq:tm)
+
+The first inequality is necessary to ensure the control task can run at the specified rate of `TASK_UPDATES_PER_SEC`.\
+The second inequality ensures that the sensors don't take up the entire timeslot.\
+The third inequality makes sure that we are able to run the control task in the allotted time slot.\
+These three combined inequalities give us both an upper and lower bound for User Event Ratio and PWM frequency relative to each other.
+
+## C-Code
+To perform the tutorial, make the following modifications to the C code you created through the [Voltage Source Inverter](/getting-started/tutorials/vsi/index.md) and [Profiling Tasks](/getting-started/tutorials/profiling-tasks/index.md) tutorials.
+
+### Set the Timing Manager Mode
+In this tutorial, we will use the [Timing Manager](/firmware/arch/timing-manager.md) in `Post-Sensor Mode`. Enable this in the `user_config.h` file by setting `USER_CONFIG_ISR_SOURCE` to `1`:
+
+`user_config.h`:
+```c
+// Specify the source of the scheduler ISR
+// Mode 0: legacy mode - scheduler is triggered based on the PWM carrier events and ratio
+//         of carrier frequency to desired control frequency
+// Mode 1: post-sensor mode - scheduler is triggered when all the enabled sensors are done
+//         acquiring their data
+
+#define USER_CONFIG_ISR_SOURCE (1)
+```
+
+### Link the Timing Manager to Sensor Interaces
+
+We need to link the sensor interfaces we wish to synchronize to with the [Timing Manager](/firmware/arch/timing-manager.md). In this tutorial, we will consider only the internal ADC (analog to digital converter) of the AMDC. However, in general, this can include other sensor peripherals, such as the encoder interface and AMDS.
+
+To link the ADC to the [Timing Manager](/firmware/arch/timing-manager.md), edit the  `...controller_init()` function to include the function call `timing_manager_enable_sensor(ADC)`:
+
+`app_controller.c`:
+```C
+void app_controller_init(void)
+{
+    // Enable data sampling for ADC
+    timing_manager_enable_sensor(ADC);
+    // Register "ctrl" command with system
+    cmd_controller_register();
+}
+```
+
```

Comment:
Have them add new functions to report out `acquisition time` and `time since done` based on a command line command.